packages/field-base/src/char-length-mixin.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+import { ForwardInputPropsMixin } from './forward-input-props-mixin.js';
+
+const CharLengthMixinImplementation = (superclass) =>
+  class CharLengthMixinClass extends ForwardInputPropsMixin(superclass) {
+    static get properties() {
+      return {
+        /**
+         * Maximum number of characters (in Unicode code points) that the user can enter.
+         */
+        maxlength: {
+          type: Number
+        },
+
+        /**
+         * Minimum number of characters (in Unicode code points) that the user can enter.
+         */
+        minlength: {
+          type: Number
+        }
+      };
+    }
+
+    static get forwardProps() {
+      return [...super.forwardProps, 'maxlength', 'minlength'];
+    }
+
+    static get constraints() {
+      return [...super.constraints, 'maxlength', 'minlength'];
+    }
+
+    /**
+     * Override an observer from `ValidateMixin` to add `minlength` and `maxlength` constraints.
+     * @param {boolean | undefined} required
+     * @param {number | undefined} minlength
+     * @param {number | undefined} maxlength
+     * @protected
+     * @override
+     */
+    _constraintsChanged(required, minlength, maxlength) {
+      // Prevent marking field as invalid when setting required state
+      // or any other constraint before a user has entered the value.
+      if (!this.invalid) {
+        return;
+      }
+
+      if (required || minlength || maxlength) {
```

Comment:
The case when `minlength = 0` or `maxlength = 0` doesn't run validation with the current condition.

Reply:
This is now handled by `InputConstraintsMixin` and there are two tests for `0` values.